src/bootstrap/doc.rs

```diff
@@ -263,9 +258,7 @@ fn invoke_rustdoc(builder: &Builder, target: Interned<String>, markdown: &str) {
         t!(t!(File::create(&version_info)).write_all(info.as_bytes()));
     }
 
-    let mut cmd = Command::new(&rustdoc);
-
-    builder.add_rustc_lib_path(compiler, &mut cmd);
+    let mut cmd = builder.rustdoc_cmd(builder.compiler(0, build.build));
```

Comment:
Interestingly, not without the full bootstrap option enabled. `./x.py doc` requests two things: `compile::Rustdoc { target_compiler: stage = 1 }` and `stage = 0`. However, both of those call into the same `ToolBuild`, constructing a "stage 1" rustdoc (there's not really a way to make a stage 0 rustdoc, so we just pretend that what you want is a stage1 rustdoc). I'll still add the threading code, though.